control/src/lib.rs

```diff
@@ -203,6 +205,10 @@ pub mod pallet {
 	#[pallet::generate_deposit(pub(super) fn deposit_event)]
 	// SBP-M2 review: I would suggest limiting events' attributes.
 	// When someone is interested, there should be a possibility to query current state.
+	
+	// vovacha: This is done only for indexing purposes, so the indexer would get everythong from the event 
+	// instead of quering the chain directly.
+	// Maybe we should reconsider it, but currently there are no obvious reasons of removing those fields
```

Comment:
i think that is exactly the point, to minimise data in the event and query data directly from the chain.

Reply:
If we want to query the chain's storage directly, what's the purpose of caching the chain with the indexer?  I've already had this discussion with the Subsquid team   and with the guys at substrate StackExchange.

>   the main downside of Events with 10-20 fields is that each event incurs some storage overhead and computational overhead during cleanup. In practice, as Events are themselves stored (temporarily) in the state of the chain, there is little meaningful difference between emitting larger events versus emitting more smaller events.

<img width="575" alt="squid" src="https://user-images.githubusercontent.com/16227101/201426095-cb8dc90f-87e5-4e74-a2b0-4eb4e1035252.png">
  

    I'm okay with any approach, but we should decide on something.

Reply:
Is the issue only with the space that events are consuming in blockchain?
If so, we just can estimate the size of each event and its frequency to understand the impact on block size.
E.g. for the control pallet most heavy events like `OrgCreated`, `OrgUpdated` will not be used frequently I suppose. `MemberAdded` and `FundsSpended` will be called more frequently, but they have 3-5 arguments, which is fine I guess.

Reply:
https://docs.substrate.io/build/events-and-errors/

Reply:
I am pretty sure we are about to start querying chain storage during indexing process at some stage. The reason of doing so, is that we don't only rely on events from GameDAO pallets, which we are free to update however we want and include as much event details as needed.
But we are also going to use other pallets, built in FRAME or part of ORML stack. Those pallets not always contain needed details in the event. For example, event generated after updating the identity does not contain actual identity information.

So we have two possible solutions of given problem:
1. Patch every pallet, where we need extended event information.
2. Query information from the chain.

I would prefer second option.